src/Kentico.Xperience.Lucene/Services/ILuceneTaskProcessor.cs

```diff
@@ -15,6 +18,7 @@ public interface ILuceneTaskProcessor
     /// <param name="queueItems">The items to process.</param>
     /// <param name="cancellationToken">The cancellation token for the task.</param>
     /// <returns>The number of items processed.</returns>
-    Task<int> ProcessLuceneTasks(IEnumerable<LuceneQueueItem> queueItems, CancellationToken cancellationToken);
+    
+    int ProcessLuceneTasks(IEnumerable<LuceneQueueItem> queueItems, CancellationToken cancellationToken, int maximumBatchSize = 100);
```

Comment:
```suggestion
Task<int> ProcessLuceneTasks(IEnumerable<LuceneQueueItem> queueItems, CancellationToken cancellationToken, int maximumBatchSize = 100);
```

If a method accepts a `CancellationToken` it should return a `Task`.

Reply:
Changed